### ISSUE TITLE
STG stuck in infinite loop

### ISSUE THREAD

**Issue by nextghost** [Drive4ik/simple-tab-groups#735]
**Describe the bug**
Yesterday, STG 4.7 got stuck in an infinite "loading" loop and Firefox became unresponsive. Restarting Firefox did not help, STG would immediately get stuck in the loading loop again and no tab content would load. I had to quickly open addons manager after starting Firefox, disable STG, remove it completely and downgrade to 4.6.2, then restore tab groups from backup.

**To Reproduce**
I cannot reproduce this bug at will. However, the issue was likely triggered by having two Firefox windows open at the same time and closing one of them. When this bug happened, some tabs from the closed window "jumped" to the other window that remained open.

**Expected behavior**
Working browser with tab groups.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS and version: Gentoo Linux, fully updated on Sunday
 - Firefox version: 83.0
 - Simple Tab Groups version: 4.7 (not sure if earlier versions are also affected)

**Additional context**
N/A

**Comment by Moilleadoir** [comment]
I’m having a similar problem which happened after closing the wrong Firefox window. I think something about the groups data might now be invalid. I’ve tried removing the session store, but tabs still load. Is there any way to delete the config?

**Comment by nextghost** [comment]
Here's my workaround which resolved the issue for me so far:
- Open Firefox
- Press Ctrl+Shift+A before STG has time to start
- Disable STG
- Uninstall STG
- Restart Firefox
- Go to https://addons.mozilla.org/en-US/firefox/addon/simple-tab-groups/versions/
- Install STG version 4.6.2
- Disable automatic updates for STG
- Restore tab groups from backup. The last automatic backup that'll load was probably November 23rd or the day before.

**Comment by grahamperrin** [comment]
Hi

I have used Simple Tab Groups `4.7` for three months with more than a thousand tabs and around a hundred other extensions enabled, with nothing to suggest that it's buggy in the ways described above. 

https://github.com/Drive4ik/simple-tab-groups/issues/735#issue-751843088

> … two Firefox windows … closing one … tabs from the closed window "jumped" to the other …

This is expected behaviour. 

Appearance in the window behind the window that was closed is followed by _disappearance_ of the tabs – they'll be hidden by Firefox (as instructed by Simple Tab Groups). 

The time taken for Firefox to hide each tab will depend upon the total number of tabs (hidden and non-hidden) in the session, and other factors. 

Did you force Firefox to quit at any time after closing the window? Have you subsequently forced Firefox to quit at any time?

I wonder whether your installation of Firefox is an underlying trouble. 

### [Troubleshoot Firefox issues using Safe Mode | Firefox Help](http://mzl.la/1xKrE7R)

With safe mode it's easy to disable, or remove, an extension. No need to rush. 

Also, it's sometimes found that Firefox will be more reliable through a simple start in safe mode followed by a normal start.

**Comment by nextghost** [comment]
> Did you force Firefox to quit at any time after closing the window? Have you subsequently forced Firefox to quit at any time?

Yes. I didn't have much choice, given that Firefox (both main process and webextension subprocess together) ate all available memory and basically froze in a swapping loop for more than 30 minutes. If the problem is that STG config file got corrupted by forced exit, could you maybe write the config into a temporary file and then rename it at the end? Rename is an atomic operation so it should prevent data corruption.
